test/test_copy.cc

```diff
@@ -171,30 +133,14 @@ void test_copy_work(Params& params, bool run)
         #ifdef SLATE_HAVE_SCALAPACK
             // comparison with reference routine from ScaLAPACK
 
-            // BLACS/MPI variables
-            blas_int ictxt, p_, q_, myrow_, mycol_;
-            blas_int A_desc[9], B_desc[9];
-            blas_int mpi_rank_ = 0, nprocs = 1;
-
             // initialize BLACS and ScaLAPACK
-            Cblacs_pinfo(&mpi_rank_, &nprocs);
-            slate_assert( mpi_rank_ == mpi_rank );
-            slate_assert(p*q <= nprocs);
-            Cblacs_get(-1, 0, &ictxt);
-            Cblacs_gridinit(&ictxt, "Col", p, q);
-            Cblacs_gridinfo(ictxt, &p_, &q_, &myrow_, &mycol_);
-            slate_assert( p == p_ );
-            slate_assert( q == q_ );
-            slate_assert( myrow == myrow_ );
-            slate_assert( mycol == mycol_ );
-
-            int64_t info;
-            scalapack_descinit(A_desc, m, n, nb, nb, 0, 0, ictxt, lldA, &info);
-            scalapack_descinit(B_desc, m, n, nb, nb, 0, 0, ictxt, lldB, &info);
-            slate_assert(info == 0);
-
-            A_norm = slate::norm( slate::Norm::Max, A );
-            B_norm = slate::norm( slate::Norm::Max, B );
+            blas_int ictxt, A_desc[9], B_desc[9];
+            create_ScaLAPACK_context( slate::GridOrder::Col, p, q, &ictxt );
```

Comment:
I wonder if `ictxt` should be an object, which calls `Cblacs_gridexit` in its destructor.